File: HandsOn/selection_sort/selection_sort.kt
fun findSmallest(arr: MutableList<Int>): Int {
    var smallest = arr[0]
    var smallestIndex = 0
    for (i in 1 until arr.size) {
        if (arr[i] < smallest) {
            smallest = arr[i]
            smallestIndex = i
        }
    }
    return smallestIndex
}

fun selectionSort(arr: MutableList<Int>): List<Int> {
    val sortedArr = mutableListOf<Int>()
    for (i in 0 until arr.size) {
        val smallestIndex = findSmallest(arr)
        sortedArr.add(arr.removeAt(smallestIndex))
    }
    return sortedArr
}

File: HandsOn/selection_sort/selection_sort.ts
function findSmallest(arr: number[]): number {
    let smallest = arr[0];
    let smallestIndex = 0;
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallestIndex = i;
        }
    }
    return smallestIndex;
}

function selectionSort(arr: number[]): number[] {
    let sortedArr: number[] = new Array(arr.length);
    for (let i = 0; i < arr.length; i++) {
        let smallestIndex = findSmallest(arr);
        sortedArr[i] = arr[smallestIndex];
        arr[smallestIndex] = Number.MAX_VALUE;
    }
    return sortedArr;
}
